lib/server.js

```diff
@@ -318,7 +318,8 @@ class Server extends EventEmitter {
     const ident = (cfg.ident ? Buffer.from(cfg.ident) : undefined);
     const offer = new KexInit(algorithms);
 
-    this._srv = new netServer((socket) => {
+
+    this.newClient = (socket) => {
```

Comment:
Rather than creating a new property, it'd be better to just move the code into the `'connection'` handler.

Reply:
That will make the external API complex, i’ll need to hook the « connection » event, unsubscribe ssh2, wait for a socket, detect the proxy-protocol header and emit ssh2  connection listener manually…

What about an async property « validateSocket(socket, cb) » with the current (and undocumented) maxClient logic that users might want to override ?

Reply:
I was assuming someone would pass a fake server object that they would then emit `'connection'` on as needed. It would more code changes, but we could reduce the `net.Server`-specific API usage (e.g. `listen()`) in that case so the end user wouldn't need to implement/stub out as much.

Reply:
What’s the getConnections API purpose here ? Can we drop it ?

Reply:
It's to provide access to the underlying server's [`getConnections()`](https://nodejs.org/docs/latest/api/net.html#net_server_getconnections_callback), which gets you the number of connected sockets.

I would rather not drop it, especially since most people who are implementing a server are most likely using the implicit `net.Server` for listening for incoming connections. If anything we could just check if the function exists on the underlying server instance. If it doesn't exist or isn't a function, then we could probably just throw an exception I suppose.

Reply:
i’ve submitted a new implementation